pr_testing/test-cmssw-provides.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -xe
+if [ $# -ne 6 ]; then
+  echo Usage: test-cmssw-provides.sh scram_arch pkgtools_branch cmsdist_tag build_dir week_num cmssw_release
+  exit 1
+fi
+
+SCRAM_ARCH=$1
+PKG_TOOL_BRANCH=$2
+CMSDIST_TAG=$3
+BUILD_DIR=$4
+WEEK_NUM=$5
+CMSSW_RELEASE=$6
+
+CMSSW_RELEASE_BASE=${CMSSW_RELEASE}
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"  # Absolute path to script
+CMS_BOT_DIR=$(dirname ${SCRIPTPATH})  # To get CMS_BOT dir path
+
+rm -rf $WORKSPACE/test-provides
+mkdir -p $WORKSPACE/test-provides
+cd $WORKSPACE/test-provides
+if [ ! -d $WORKSPACE/pkgtools ]; then
+  git clone git@github.com:cms-sw/pkgtools --depth 1 -b $PKG_TOOL_BRANCH
+else
+  ln -s $WORKSPACE/pkgtools .
+fi
+
+if [ ! -d $WORKSPACE/cmsdist ]; then
+  git clone git@github.com:cms-sw/cmsdist --depth 1 -b $CMSDIST_TAG
+else
+  ln -s $WORKSPACE/cmsdist .
+fi
+
+if [ -d ${WORKSPACE}/$BUILD_DIR/$SCRAM_ARCH/var/lib/rpm ]; then
+  RPM_CMD="/cvmfs/cms-ib.cern.ch/sw/$(uname -m)/week${WEEK_NUM}/common/cmspkg -a $SCRAM_ARCH env -- rpm"
+  if [ ! -z ${CMSSW_FULL_RELEASE_BASE} ]; then
+    CMSSW_RELEASE_PATCH=$CMSSW_RELEASE
+    CMSSW_RELEASE_BASE=$(echo ${CMSSW_FULL_RELEASE_BASE} | rev | cut -d '/' -f 1 | rev)
+  fi
+
+  PROVIDELIST=$(${RPM_CMD} -q --provides cms+cmssw+${CMSSW_RELEASE_BASE})
+  PROVIDELIST=$(echo $PROVIDELIST | sed -E 's/^(.*)$/Provides: \1/g')
+
+  if [ ! -z ${CMSSW_RELEASE_PATCH} ]; then
+    PROVIDELIST_PATCH=$(${RPM_CMD} -q --provides cms+cmssw-patch+${CMSSW_RELEASE_PATCH})
+    PROVIDELIST_PATCH=$(echo ${PROVIDELIST_PATCH} | sed -E 's/^(.*)$/Provides: \1/g')
+    PROVIDELIST="$PROVIDELIST\n${PROVIDELIST_PATCH}"
```

Comment:
@iarspider , looks like provides for patch release are not correctly added. I tested it locally and I see
```
Singularity> ./common/cmspkg -a el8_amd64_gcc11 env -- rpm -q --provides cms+cmssw-pr-package+1.0
cms+cmssw+CMSSW_13_1_X_2023-03-01-1100
cms+cmssw+CMSSW_13_1_X_2023-03-01-1100 = 1-1
cms+cmssw+CMSSW_13_1_X_2023-03-01-1100(x86-64) = 1-1
cms+cmssw-patch+CMSSW_13_1_X_2023-03-01-2300
cms+cmssw-patch+CMSSW_13_1_X_2023-03-01-2300 = 1-1
cms+cmssw-patch+CMSSW_13_1_X_2023-03-01-2300(x86-64) = 1-1
cms+cmssw-pr-package+1.0
cms+cmssw-pr-package+1.0 = 1-1
cms+cmssw-pr-package+1.0(x86-64) = 1-1
libAlignmentCocoaDDLObjects.so()(64bit)
libAlignmentCocoaDaq.so()(64bit)
....
```
which is fine but at the end I see
```
libtestDTCalibration.so()(64bit)
libtestSimMuonDTDigitizer.so()(64bit)
perl(Mpslib) = 1.01
perl(SCRAMGenUtils)\nProvides:
```
You do not have to join provides of cmssw and cmssw-patch in single line. You can just dump thses separately e.g
for cmssw
```
${RPM_CMD} -q --provides cms+cmssw+${CMSSW_RELEASE_BASE}  | sed -E 's/^(.*)$/Provides: \1/g' > cmsdist/cmssw-pr-provides.file
```
for cmssw-patch
```
${RPM_CMD} -q --provides cms+cmssw-patch+${CMSSW_RELEASE_PATCH}  | sed -E 's/^(.*)$/Provides: \1/g' >> cmsdist/cmssw-pr-provides.file
```
should be enough to register the provides